plugins/ui/src/js/src/elements/UITable/JsTableProxy.ts

```diff
@@ -0,0 +1,100 @@
+import type { dh } from '@deephaven/jsapi-types';
+
+export interface UITableLayoutHints {
+  frontColumns?: string[];
+  frozenColumns?: string[];
+  backColumns?: string[];
+  hiddenColumns?: string[];
+  columnGroups?: dh.ColumnGroup[];
+}
+
+// This tricks TS into believing the class extends dh.Table
+// Even though it is through a proxy
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface JsTableProxy extends dh.Table {}
+
+/**
+ * Class to proxy JsTable.
+ * Any methods implemented in this class will be utilized over the underlying JsTable methods.
+ * Any methods not implemented in this class will be proxied to the table.
+ */
+class JsTableProxy {
+  private table: dh.Table;
+
+  layoutHints: dh.LayoutHints | null = null;
+
+  constructor({
+    table,
+    layoutHints,
+  }: {
+    table: dh.Table;
+    layoutHints: UITableLayoutHints;
+  }) {
```

Comment:
Any reason not to make `layoutHints` optional?
```suggestion
  constructor({
    table,
    layoutHints = {},
  }: {
    table: dh.Table;
    layoutHints?: UITableLayoutHints;
  }) {
```

Reply:
Right now we only need it from 1 spot, so I didn't want it to get accidentally left out if we had to create the table conditionally in the future for some reason. At least this way you have to be explicit that you're dropping `layoutHints`

I guess we could make it `| undefined` and then make undefined not set the value in the proxy class so it falls back to the underlying table. But that seems unnecessary right now